gpii/node_modules/singleInstance/src/singleInstance.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Ensures there's only a single instance of GPII.
+ *
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The R&D leading to these results received funding from the
+ * Department of Education - Grant H421A150005 (GPII-APCP). However,
+ * these results do not necessarily represent the policy of the
+ * Department of Education, and you should not assume endorsement by the
+ * Federal Government.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("universal");
+var fs = require("fs");
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.singleInstance");
+
+/**
+ * Declares this instance of GPII to be the only instance, unless there's already one running.
+ *
+ * The "pid/lock file" method is used, where a file containing the pid of the current process is created. If it
+ * already exists, and the pid points to a running GPII instance, then the lock will fail.
+ *
+ * File IO is performed synchronously, ensuring the checks are made before GPII does anything else (eg, bind to a port).
+ *
+ * @todo More checks need to be made to ensure the pid is GPII, and not just another process.
+ *
+ * @param pidFile {String} [Optional] The path of the pid file. Default is "gpii.pid" in the settings directory.
+ * @return {boolean} true if this instance is the only GPII instance.
+ */
+gpii.singleInstance.registerInstance = function (pidFile) {
+    if (!pidFile) {
+        var settingsDirComponent = gpii.settingsDir();
+        pidFile = settingsDirComponent.getGpiiSettingsDir() + "/gpii.pid";
+    }
```

Comment:
It's annoying, but the settingsDir component will need to be explicitly disposed here using settingsDirComponent.destroy() otherwise its gunk will hang around in memory indefinitely